### PR TITLE
Add JOOQ and Flyway

### PR DESCRIPTION
**Pull request by mmthatsagoodcoffee**
# Description:

Added JOOQ and Flyway to dependencies. Added configuration. Both are technically working, however it's not done. It will require decent documentation for sure, possibly a cleanup. I'm waiting for suggestions as well.

# Checklist:
Tests have been added or updated to cover the changes.
Code follows the team's coding conventions and style guidelines.
Documentation has been updated (if applicable).
The branch has been tested and is ready to be merged.

# Additional Notes:
If you want to check it out setup a postgres in a docker container, make sure username is "postgres" and password is "admin", port is 5432 and create a database "backendtemplate" there. Next, run ./gradlew flywayMigrate and /gradlew generateJooq when first one suceeds :) Flyway should add table users to your database, and JOOQ will show it in /generated folder as its generated classes.

### CODE REVIEW

**Review by Chris-55 — APPROVED** [review]
LGTM

File: dbschema/src/main/resources/db/migration/V2__create_table_users.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE "users" (
+    "id" SERIAL PRIMARY KEY,
+    "public_id" UUID NOT NULL DEFAULT uuid_generate_v4(),
+    "created" TIMESTAMP NOT NULL DEFAULT NOW(),
+    "email" TEXT NOT NULL,
+    "passwordHash" TEXT,
+    "name" TEXT,
```
Comment (qxtno) [inline dbschema/src/main/resources/db/migration/V2__create_table_users.sql:7]: Doesn't it make sense to make these two columns not null as well?

Comment (mmthatsagoodcoffee) [reply dbschema/src/main/resources/db/migration/V2__create_table_users.sql:7]: Possibly yeah, but I only did that to proof it works didn't really think anything through

Comment (qxtno) [reply dbschema/src/main/resources/db/migration/V2__create_table_users.sql:7]: Okay, we can change it later if we need to

**Review by qxtno — APPROVED** [review]

**Pull request merged**